ui/src/View.elm

```diff
@@ -24,20 +23,23 @@ view model =
 
 
 mainView : Model -> Html Msg
-mainView { loading, error, buildsPager, filters, filterValues, settings } =
+mainView { loading, settings, error, buildsPager, facets, page, filters, filterValues } =
     div [ class "row" ]
         [ div [ class "col-sm-9" ]
-            (if loading then
-                [ spinner ]
-             else
-                [ errorView error
-                , numBuilds buildsPager
-                , div [] <| List.map recordView buildsPager.objects
-                , nextPageBtn settings.pageSize (List.length buildsPager.objects) buildsPager.total
-                ]
-            )
+            [ errorView error
+            , case facets of
+                Just facets ->
+                    div []
+                        [ paginationView facets settings.pageSize page
+                        , div [] <| List.map recordView facets.hits
+                        , paginationView facets settings.pageSize page
+                        ]
+
+                Nothing ->
+                    spinner
+            ]
         , div [ class "col-sm-3" ]
-            [ filtersView filters filterValues
+            [ filtersView facets filters filterValues
```

Comment:
I'm not crazy about `filtersView` having to take a `Maybe` for facets. Maybe do another `case` here to put a spinner here too?